runner/src/main.rs

```diff
@@ -600,10 +656,69 @@ struct Example {
     clients: HashMap<String, Executable>,
 }
 
+impl Example {
+    fn get_application(&self, application_variant: &str) -> &Application {
+        self.applications
+            .get(application_variant)
+            .unwrap_or_else(|| panic!(
+                "Unsupported application variant: {} (supported variants include: all, rust, cpp, go, nodejs, none)",
+                application_variant)
+            )
+    }
+
+    fn construct_backend_build_steps(&self, build_client: &BuildClient) -> Vec<Step> {
+        self.backends
+            .iter()
+            .map(|(name, backend)| Step::Single {
+                name: name.to_string(),
+                command: build(&backend.target, build_client),
+            })
+            .collect()
+    }
+
+    /// Recursively constructs backend steps.
+    fn construct_backend_run_steps(
+        &self,
+        run_server_clients: Step,
+        build_client: &BuildClient,
+    ) -> Step {
+        self.backends
+            .iter()
+            // First iteration includes `run_server_clients` as a foreground step.
+            .fold(run_server_clients, |backend_steps, (name, backend)| {
+                Step::WithBackground {
+                    name: name.to_string(),
+                    // Each `backend` is included as background step.
+                    background: run(&backend, &build_client, Vec::new()),
+                    foreground: Box::new(backend_steps),
+                }
+            })
+    }
+}
+
+#[derive(serde::Deserialize, Debug, PartialEq)]
+#[serde(deny_unknown_fields)]
+enum ExampleType {
+    // A classic Oak Example that needs to run with the `oak_loader`
+    Classic,
+    // An Oak-Functions example that needs to run with `oak_functions_loader`
+    OakFunctions,
+}
+
+impl Default for ExampleType {
+    fn default() -> Self {
+        ExampleType::Classic
+    }
+}
+
 #[derive(serde::Deserialize, Debug)]
 #[serde(deny_unknown_fields)]
 struct Application {
+    // In case of classic Oak applications, this is the path to a manifest file.
+    // In case of Oak functions applications, this is an empty string.
     manifest: String,
+    // In case of classic Oak applications, this is the path to the `.oak` application file.
+    // In case of Oak functions applications, this is the path to the Wasm module.
     out: String,
     modules: HashMap<String, Target>,
 }
```

Comment:
Have you considered specializing this in ApplicationClassic and ApplicationFunction, instead of using a string at the top level to determine how to interpret the fields in this struct?